src/Microsoft.NetCore.Analyzers/Core/Security/UseSharedAccessProtocolHttpsOnly.cs

```diff
@@ -0,0 +1,140 @@
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.ValueContentAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class UseSharedAccessProtocolHttpsOnly : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "CA5376";
+        private static readonly LocalizableString s_Title = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.UseSharedAccessProtocolHttpsOnly),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Message = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.UseSharedAccessProtocolHttpsOnlyMessage),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.UseSharedAccessProtocolHttpsOnlyDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var microsoftWindowsAzureStorageNamespaceSymbol = compilationStartAnalysisContext
+                            .Compilation
+                            .GlobalNamespace
+                            .GetMembers("Microsoft")
+                            ?.FirstOrDefault()
+                            .GetMembers("WindowsAzure")
+                            .OfType<INamespaceOrTypeSymbol>()
```

Comment:
>. [](start = 28, length = 1)

Do you need ?. here also, like the after the first GetMembers() call?

Can you add some unit tests for various combinations of namespaces defined?

Reply:
Actually GetMembers can never return null but you need conditional access (?.) after all the FirstOrDefault invocations

Reply:
Please add some unit tests that define these namespaces (but not "Microsoft.WindowsAzure.Storage"):
- Microsoft
- Microsoft.WindowsAzure

.FirstOrDefault() will return null for some of those cases, right?